core/src/test/java/org/opensearch/sql/expression/operator/predicate/BinaryPredicateOperatorTest.java

```diff
@@ -209,13 +209,31 @@ private static Stream<Arguments> testLikeArguments() {
     return builder.build();
   }
 
+  private static Stream<Arguments> getCidrArguments() {
+    return Stream.of(
+            Arguments.of("10.24.33.5", "10.24.34.0/24", false), // IPv4 less
+            Arguments.of("10.24.34.5", "10.24.34.0/24", true),   // IPv4 between
+            Arguments.of("10.24.35.5", "10.24.34.0/24", false), // IPv4 greater
+            Arguments.of("10.24.35.5", "10.24.34.0/33", false), // IPv4 prefix too long
+
+            Arguments.of("2001:0db7::8329", "2001:0db8::/32", false), // IPv6 less
+            Arguments.of("2001:0db8::8329", "2001:0db8::/32", true),  // IPv6 between
+            Arguments.of("2001:0db9::8329", "2001:0db8::/32", false), // IPv6 greater
+            Arguments.of("2001:0db9::8329", "2001:0db8::/129", false), // IPv6 prefix too long
+
+            Arguments.of("INVALID", "10.24.34.0/24", false), // Invalid address
+            Arguments.of("10.24.34.5", "INVALID", false), // Invalid range
+            Arguments.of("10.24.34.5", "10.24.34.0/INVALID", false) // Invalid prefix
+    );
+  }
+
   @ParameterizedTest(name = "and({0}, {1})")
   @MethodSource("binaryPredicateArguments")
   public void test_and(Boolean v1, Boolean v2) {
     FunctionExpression and = DSL.and(DSL.literal(booleanValue(v1)), DSL.literal(booleanValue(v2)));
     assertEquals(BOOLEAN, and.type());
     assertEquals(v1 && v2, ExprValueUtils.getBooleanValue(and.valueOf(valueEnv())));
-    assertEquals(String.format("and(%s, %s)", v1.toString(), v2.toString()), and.toString());
+    assertEquals(String.format("and(%s, %s)", v1, v2.toString()), and.toString());
```

Comment:
This and the adding of final are good changes, but we should keep changes in this PR specific to the issue you're working on. Let's have the clean-up improvements in a separate Issue and PR.

Reply:
Sounds good. I've reverted the cleanup changes to `final` and `toString`. ✅